core/store/src/trie/mod.rs

```diff
@@ -586,6 +586,130 @@ impl Trie {
         Ok(())
     }
 
+    // Prints the trie nodes starting from hash, up to max_depth depth.
+    pub fn print_recursive(&self, hash: &CryptoHash, max_depth: u32) -> String {
+        let mut buf = String::new();
+        let mut prefix: Vec<u8> = Vec::new();
+        self.print_recursive_internal(&mut buf, hash, max_depth, &mut "".to_string(), &mut prefix)
+            .expect("printing failed");
+        buf
+    }
+
+    // Converts the list of Nibbles to a readable string.
+    fn nibbles_to_string(&self, prefix: &Vec<u8>) -> String {
+        let mut result = String::new();
+        for i in (0..prefix.len()).step_by(2) {
+            if i + 1 < prefix.len() {
+                let chr: char = ((prefix[i] * 16) + prefix[i + 1]).into();
+                result.push_str(&format!("{}", chr.escape_default()));
+            }
+        }
+        if prefix.len() % 2 == 1 {
+            result.push_str(&format!(" + {}", prefix[prefix.len() - 1]));
+        }
+        result
+    }
+
+    fn print_recursive_internal(
+        &self,
+        f: &mut dyn std::fmt::Write,
+        hash: &CryptoHash,
+        max_depth: u32,
+        spaces: &mut String,
+        prefix: &mut Vec<u8>,
+    ) -> std::fmt::Result {
+        if max_depth == 0 {
+            writeln!(
+                f,
+                "{}max depth reached hash:{} prefix:{}",
+                spaces,
+                hash,
+                self.nibbles_to_string(prefix)
+            )?;
+            return Ok(());
+        }
```

Comment:
This feels like it would be a noisy message.  The hash and prefix had already been printed by the caller and we can see that max depth was reached by the fact that nothing deeper is printed. ;)

Reply:
good point.